netcore/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -20,27 +20,9 @@ private class RawData
 			public byte Data;
 		}
 
-		public int Length {
-			get {
-				int length = GetLength (0);
-
-				for (int i = 1; i < Rank; i++) {
-					length *= GetLength (i);
-				}
-				return length;
-			}
-		}
+		public int Length => checked ((int)Unsafe.As<RawData>(this).Count);
```

Comment:
I guess you will need to remove the intrinsic then 🙂

Reply:
Oh right.

Reply:
Why only netcore?

Reply:
The motivation for this was https://github.com/dotnet/coreclr/pull/27123. I was testing the grounds to try to see if we can share more of the code with CoreCLR eventually. The only reason to port it to non-netcore is code parity and removing the native code altogether. It doesn't generate better or worse code in the end so I am not sure if it's worth it.